pkg/executor/bacalhau/client.go

```diff
@@ -1,73 +1,269 @@
 package bacalhau
 
 import (
-	"encoding/json"
+	"context"
 	"fmt"
-	"io"
+	"strings"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
 
 	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
-	retryablehttp "github.com/hashicorp/go-retryablehttp"
+	clientv2 "github.com/bacalhau-project/bacalhau/pkg/publicapi/client/v2"
+	"github.com/lilypad-tech/lilypad/pkg/data"
+	"github.com/lilypad-tech/lilypad/pkg/data/bacalhau"
 )
 
-const API_VERSION = "v1"
-
 type BacalhauClient struct {
-	http    *retryablehttp.Client
-	options BacalhauExecutorOptions
+	api clientv2.API
 }
 
-func newClient(options BacalhauExecutorOptions) (*BacalhauClient, error) {
-	http := retryablehttp.NewClient()
-	return &BacalhauClient{
-		http:    http,
-		options: options,
-	}, nil
+func NewBacalhauClient(apiHost string) (*BacalhauClient, error) {
```

Comment:
Why are we making these functions public? Do we plan to use them outside this package?